### ISSUE TITLE
REQ: Browse source DIR

### ISSUE THREAD

**Issue by quadcom** [tamland/kodi-addon-extras#8]
Pretty quiet around here eh?

I was hoping you could consider adding the ability to just browse the source DIR instead of a folder in that DIR. 

Radarr-Extended pulls in trailers and other extras. But you can't define where they are saved. It just puts them next to the primary object.

I tried to set the DIR in settings to " " but it won't take, and the debug logs throw an error if you manually edit the config file.

Pretty please!!!

Thx

**Comment by pkscout** [comment]
That's not going to be an option.  This add-on basically just lists everything in a given directory, and if everything is in the root of the directory then the movie/tvshow episodes would get listed with everything else.  If you follow the instructions, you'll see that you need to exclude the extras directory from the content scan, otherwise Kodi will keep trying to add all the extra content as a movie or TV show.  If the extra content isn't in a separate folder, there's no way to do that, and you run the risk of having random things added to your library.

You should ask the Radarr-Extended folks to provide an option to pull in all those things in a subdirectory.

**Comment by quadcom** [comment]
So I did add the exclude regex in my advancedsettings. But taking a look at it, I do see something that may be causing the problem for me.

```
<video>
  <excludefromscan>
       <regexp>[-\._ ](extrafanart|sample|trailer|extrathumbs)[-\._ ]</regexp>
  </excludefromscan>
  <excludefromlisting>
       <regexp>[-._ \\/](extrafanart|sample|trailer|extrathumbs)[-._ \\/]</regexp>
  </excludefromlisting>
  <!-- Extras: Section Start -->
       <excludefromscan action="append">
       <regexp>/extras/</regexp>
       <regexp>[\\/]extras[\\/]</regexp>
  </excludefromscan>
  <excludetvshowsfromscan action="append">
        <regexp>/extras/</regexp>
        <regexp>[\\/]extras[\\/]</regexp>
  </excludetvshowsfromscan>
  <!-- Extras: Section End -->
</video>
```

I have `excludefromlisting` with ''trailers'' and ''samples''. Now that I think about it, if the word trailers is there it would be excluded from listing, duh!

That now reads 

```
<video>
	<excludefromscan>
		<regexp>[-\._ ](extrafanart|sample|trailer|extrathumbs)[-\._ ]</regexp>
	</excludefromscan>
	<excludefromlisting>
		<regexp>[-._ \\/](extrafanart|extrathumbs)[-._ \\/]</regexp>
	</excludefromlisting>
	<!-- Extras: Section Start -->
	<excludefromscan action="append">
		<regexp>/extras/</regexp>
		<regexp>[\\/]extras[\\/]</regexp>
	</excludefromscan>
	<excludetvshowsfromscan action="append">
		<regexp>/extras/</regexp>
		<regexp>[\\/]extras[\\/]</regexp>
	</excludetvshowsfromscan>
	<!-- Extras: Section End -->
</video>
```

I've pulled those out of the config, and I'll test it out when the wife isn't using the system!

**Comment by quadcom** [comment]
Another thing I noticed is Kodi is looking for the trailer file next to the media file, at least according to the docs. 

Does the Extras Add-on modify this functionality? 

I have searched high and low for a simple way to get to the media storage location from the objects info page, or even from the listing page using a keyboard shortcut. But I haven't found anything that does that. 

I even looked into writing something myself. But that's difficult since I haven't coded anything other than websites before.

Any suggestions?
